packages/core-mobile/app/services/notifications/NotificationsService.ts

```diff
@@ -210,83 +222,90 @@ class NotificationsService {
-    }: NotificationCallbackProps) => void
-  }) => {
-    this.decrementBadgeCount(1)
+    callback: HandleNotificationCallback
+  }): Promise<void> => {
+    await this.decrementBadgeCount(1)
     if (detail?.notification?.id) {
       await this.cancelTriggerNotification(detail.notification.id)
     }
-
-    if (detail?.notification?.data?.url) {
-      callback?.({
-        url: detail.notification.data.url as string,
-        accountIndex: detail.notification.data.accountIndex as number,
-        origin: DeepLinkOrigin.ORIGIN_NOTIFICATION,
-        isDevMode: detail?.notification?.data?.isDeveloperMode === 'true'
-      })
-    }
+    callback(detail?.notification?.data)
   }
 
   handleNotificationEvent = async ({
     type,
     detail,
     callback
   }: Event & {
-    callback?: ({
-      url,
-      accountIndex,
-      origin,
-      isDevMode
-    }: NotificationCallbackProps) => void
-  }) => {
+    callback: HandleNotificationCallback
+  }): Promise<void> => {
     switch (type) {
       case EventType.DELIVERED:
-        this.incrementBadgeCount(1)
+        await this.incrementBadgeCount(1)
         break
       case EventType.PRESS:
-        this.handleNotificationPress({
+        await this.handleNotificationPress({
           detail,
           callback
         })
         break
     }
   }
 
-  cleanupNotifications = async () => {
+  cleanupNotifications = async (): Promise<void> => {
     const pendings = await notifee.getTriggerNotifications()
-    pendings.forEach(async pending => {
+    for (const pending of pendings) {
       const timestamp = fromUnixTime(
         (pending.trigger as TimestampTrigger).timestamp
       )
       if (isPast(timestamp) && pending.notification?.id) {
         await notifee.cancelTriggerNotification(pending.notification.id)
       }
-    })
+    }
   }
 
   // only for Android to obtain the notification data when the app is in the background
   // for iOS, it is handled in the onForegroundEvent PRESS event
-  getInitialNotification = async () => {
-    return notifee.getInitialNotification()
-  }
-
-  isStakeCompleteNotificationBlocked = async () => {
-    const blockedNotifications = await this.getBlockedNotifications()
-    return blockedNotifications.get(ChannelId.STAKING_COMPLETE)
+  getInitialNotification = async (
+    callback: HandleNotificationCallback
+  ): Promise<void> => {
+    const event = await notifee.getInitialNotification()
+    if (event) {
+      callback(event.notification.data)
+    }
   }
 
-  cancelAllNotifications = async () => {
+  cancelAllNotifications = async (): Promise<void> => {
     await notifee.cancelAllNotifications()
   }
 
   createChannel = async (channel: AndroidChannel): Promise<string> => {
     return notifee.createChannel(channel)
   }
+
+  displayNotification = async ({
+    channelId,
+    title,
+    body,
+    data
+  }: {
+    channelId: ChannelId
+    title: string
+    body?: string
+    data?: { [p: string]: string | object | number }
```

Comment:
use NotifidationData?

Reply:
missed it
[Fix array and data type](https://github.com/ava-labs/core-mobile/pull/1659/commits/b1fb940934027accbc3947c307add309a5f5bf41)